v3/src/components/graph/adornments/movable-line.tsx

```diff
@@ -99,19 +116,20 @@ export const MovableLine = (props: {
             screenY = yScale((pointsOnAxes.current.pt1.y + pointsOnAxes.current.pt2.y) / 2) / ySubAxesCount,
             attrNames = {x: xAttrName, y: yAttrName},
             string = equationString(slope, intercept, attrNames),
-            classFromKey = model.setClassNameFromKey(lineKey),
-            gridContainerSelector = `.graph-adornments-grid.${instanceId}`,
-            equationContainerSelector =
-              `.movable-line-equation-container${lineKey && lineKey !== '' ? `-${classFromKey}` : ''}`,
-            lineContainerClass =
-              `${gridContainerSelector} ${equationContainerSelector}`
-          select(lineContainerClass)
+            equationContainerSelector = setEquationContainerSelector(),
+            equation = select(equationContainerSelector).select('p')
+  
+          select(equationContainerSelector)
             .style('width', `${plotWidth}px`)
             .style('height', `${plotHeight}px`)
-          select(`${lineContainerClass} p`)
-            .style('left', `${screenX}px`)
-            .style('top', `${screenY}px`)
-            .html(string)
+          equation.html(string)
+          // The equation may have been unpinned from the line if the user
+          // dragged it away from the line. Only move the equation if it
+          // is still pinned.
+          if (!equation.classed('unpinned')) {
```

Comment:
As @bfinzer mentions elsewhere, this is essentially using the DOM's `unpinned` class as the model, which will prevent it from being serialized. `pinned`/`unpinned` should be a property of the cell or model, and then the class should be determined from the cell/model.